### ISSUE TITLE
Incorrect error reported for doc comment on if statement

### ISSUE THREAD

**Issue by djrenren** [rust-lang/rust#64045]
Example code ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=3391ce9f657a47424a118ad9a6941a99)):
```rust
fn main() {
    /// Accidental doc comment
    if true {
        
    }
}
```

Error:
<img width="532" alt="image" src="https://user-images.githubusercontent.com/634666/64068842-dfdf7a80-cbf2-11e9-9c66-55e1e4f2e5c4.png">

Instead this should report a *warning* about an unused doc comment like it does on statements:

```rust
/// Accidental doc comment
let x = true;
```

<img width="676" alt="image" src="https://user-images.githubusercontent.com/634666/64068860-161cfa00-cbf3-11e9-93bc-dfa71f0cdd7c.png">

**Comment by FedericoPonzi** [comment]
I've just run this on the playground, and it looks fine:
```
  Compiling playground v0.0.1 (/playground)
warning: unused doc comment
 --> src/main.rs:2:5
  |
2 |       /// Accidental doc comment
  |       ^^^^^^^^^^^^^^^^^^^^^^^^^^
3 | /     if true {
4 | |         
5 | |     }
  | |_____- rustdoc does not generate documentation for expressions
  |
  = note: `#[warn(unused_doc_comments)]` on by default

warning: 1 warning emitted

    Finished dev [unoptimized + debuginfo] target(s) in 0.42s
     Running `target/debug/playground`

```

**Comment by Dylan-DPC** [comment]
Closing this as fixed
